convert2rhel/unit_tests/utils_test.py

```diff
@@ -229,3 +240,21 @@ def test_get_rpm_path_from_yumdownloader_output(self):
 
     def test_is_rpm_based_os(self):
         assert is_rpm_based_os() in (True, False)
+
+@pytest.mark.parametrize("raise_exception", [True, False])
+def test_remove_tmp_dir(monkeypatch, raise_exception):
```

Comment:
This test can be done better. For example, you force the test to raise OSError, but here could be the bug. Let's say some other exception will be raised. I think you tests scenarios might be as 
1. You mock TMP_DIR and assign it to tmpdir fixture(https://github.com/oamg/convert2rhel/blob/main/convert2rhel/unit_tests/conftest.py#L22), then test your function and log messages (good test)
2. You mock the TMP_DIR to return some not existing path and ensure that you capture exception correctly and correct logs were emited (bad test)

Reply:
The only external factor (checking TMP_DIR on the system) is mocked out and we only check our branches. `shutil.rmtree` can raise TypeError if `TMP_DIR` is undefined/None and `FileNotFoundError` if TMP_DIR doesn't exist. The latter is not necessary to test. But the former could be tested like so:

```py
    assert len(utils.TMP_DIR) > 0
    utils.remove_tmp_dir()
    shutil_rmtree.assert_called_once_with(utils.TMP_DIR)
```

This asserts that `utils.TMP_DIR` is defined and that we call `shutils.rmtree` with it

Reply:
@SpyTec I take as you're evolving my narrative here. So when you're mocking the TMP_DIR, with tmpdir - it is under your control and now you can add files there, make it empty, or make it not exist. And ensure that you properly handling exceptions. The key here is not guessing what will be raised, but model some real world scenarios. As I remember for example that in Centos 6 there is no such exception like FileNotFound

Reply:
@SpyTec @ZhukovGreen I've updated the unit test. Please check, if I met with your ideas :)